xbmc/windowing/gbm/WinSystemGbm.cpp

```diff
@@ -40,23 +43,35 @@ CWinSystemGbm::CWinSystemGbm() :
 
 bool CWinSystemGbm::InitWindowSystem()
 {
-  if (!m_DRM.InitDrm(&m_drm, &m_gbm))
+  if (!m_DRM->InitDrm())
   {
-    CLog::Log(LOGERROR, "CWinSystemGbm::%s - failed to initialize DRM", __FUNCTION__);
+    CLog::Log(LOGERROR, "CWinSystemGbm::%s - failed to initialize Legacy DRM", __FUNCTION__);
+    m_DRM.reset();
     return false;
   }
 
-  m_nativeDisplay = m_gbm.dev;
+  if (!m_GBM->CreateDevice(m_DRM->m_fd))
+  {
+    m_GBM.reset();
```

Comment:
`m_GBM` is never reassigned, could cause issues if `InitWindowSystem` is called a second time by Kodi

Reply:
I don't really understand. I don't think that can ever happen (not now at least).